### PR TITLE
Support schema prefixes

### LINKED ISSUE #89: Support for schema prefixes for multitenancy
Currently there is no way to paginate on a different Schema dynamically.
Something we can do in Ecto, by passing the prefix option to each Repo query.

### PR DESCRIPTION
**Pull request by tschmidleithner**
Enables prefix support which is especially useful for multitenant applications.

Please note that I already implemented nearly the same functionality a few years ago but never opened a pull request for it. Now I forked the repo again with the latest state and implemented the functionality again, considering the notes from the code review by @drewolson in #90.

Fixes #89
Closes #90

### CODE REVIEW

File: config/test.exs [hunk 1]
```diff
@@ -8,4 +8,6 @@ config :scrivener_ecto, Scrivener.Ecto.Repo,
   database: "scrivener_test",
   username: System.get_env("SCRIVENER_ECTO_DB_USER") || System.get_env("USER")
 
+config :scrivener_ecto, prefixes: ["tenant_1", "tenant_2"]
```
Comment (drewolson) [inline config/test.exs]: I don't want to add config just for tests. Please use explicit values in the tests.

File: lib/scrivener/paginater/ecto/query.ex [hunk 2]
```diff
@@ -35,20 +37,23 @@ defimpl Scrivener.Paginater, for: Ecto.Query do
 
   defp entries(query, repo, page_number, _, page_size, caller, options) do
     offset = Keyword.get_lazy(options, :offset, fn -> page_size * (page_number - 1) end)
+    prefix = options[:prefix]
```
Comment (drewolson) [inline lib/scrivener/paginater/ecto/query.ex:40]: If a prefix is not provided, what is the intended behavior?

File: priv/repo/migrations/1_create_posts.exs [hunk 3]
```diff
@@ -2,12 +2,14 @@ defmodule TestRepo.Migrations.CreatePosts do
   use Ecto.Migration
 
   def change do
-    create table(:posts) do
-      add :title, :string
-      add :body, :string
-      add :published, :boolean
+    for prefix <- [nil] ++ Application.fetch_env!(:scrivener_ecto, :prefixes) do
```
Comment (drewolson) [inline priv/repo/migrations/1_create_posts.exs]: Please do not change existing migrations. Let's test this behavior against new tables only.

File: test/scrivener/paginator/ecto/query_test.exs [hunk 4]
```diff
@@ -26,20 +26,24 @@ defmodule Scrivener.Paginator.Ecto.QueryTest do
...
       %KeyValue{
         key: "key_#{i}",
         value: rem(i, 2) |> to_string
       }
-      |> Scrivener.Ecto.Repo.insert!()
+      |> insert!(opts)
     end)
   end
 
+  defp insert!(struct_or_changeset, opts) do
```
Comment (drewolson) [inline test/scrivener/paginator/ecto/query_test.exs]: I don't want to co-mingle other testing changes with this PR. Please only add a new test, no other extracts in this file.

**Review by drewolson — CHANGES_REQUESTED** [review]

File: lib/scrivener/paginater/ecto/query.ex (hunk 2 above)
Comment (tschmidleithner) [reply lib/scrivener/paginater/ecto/query.ex:40]: If a prefix is not provided, `options[:prefix]` would return `nil` and therefore forwarded as `prefix: nil` in options for `repo.all/2`. Thus, `prefix` would simply be ignored by the Ecto query if its value is `nil`, see also https://github.com/elixir-ecto/ecto/blob/f016ccc2cffbd1ee2c3bd265dba42baee8143889/lib/ecto/query.ex#L368.

Would you prefer a different approach? E.g. removing prefix from the `repo.all/2` argument options if the value is `nil`?

File: test/scrivener/paginator/ecto/query_test.exs (hunk 4 above)
Comment (tschmidleithner) [reply test/scrivener/paginator/ecto/query_test.exs]: Good point, moving the test to an extra test file. What do you think?

File: lib/scrivener/paginater/ecto/query.ex (hunk 2 above)
Comment (drewolson) [reply lib/scrivener/paginater/ecto/query.ex:40]: Ah, ok. As long as `prefix: nil` is ignored I’m ok with this behavior.

**Review by drewolson — APPROVED** [review]

**Pull request merged**